File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/IExpediaRapidAPI.vb
Imports Intuitive.Net
Imports ThirdParty.Abstractions.Models.Property.Booking
Imports ThirdParty.VBSuppliers.ExpediaRapid.SerializableClasses

Namespace ExpediaRapid

    Public Interface IExpediaRapidAPI
        Function GetDeserializedResponse(Of TResponse As {IExpediaRapidResponse, New})(propertyDetails As PropertyDetails, request As WebRequests.Request) As TResponse
        Function GetResponse(propertyDetails As PropertyDetails, request As WebRequests.Request) As String
    End Interface

End Namespace

File: src/libs/ThirdParty.CSSuppliers/ExpediaRapid/IExpediaRapidAPI.cs
namespace ThirdParty.CSSuppliers.ExpediaRapid
{
    using Intuitive.Net.WebRequests;
    using System.Threading.Tasks;
    using ThirdParty.CSSuppliers.ExpediaRapid.SerializableClasses;
    using ThirdParty.Models.Property.Booking;

    public interface IExpediaRapidAPI
    {
        Task<TResponse> GetDeserializedResponseAsync<TResponse>(PropertyDetails propertyDetails, Request request) where TResponse : IExpediaRapidResponse<TResponse>, new();
        Task<string> GetResponseAsync(PropertyDetails propertyDetails, Request request);
    }
}